device/bluetooth_mapper/watcher/watcher.go

```diff
@@ -100,8 +100,21 @@ func onPeripheralDisconnected(p gatt.Peripheral, err error) {
 //onPeripheralConnected contains the operations to be performed as soon as the peripheral device is connected
 func (w *Watcher) onPeripheralConnected(p gatt.Peripheral, err error) {
 	actionmanager.GattPeripheral = p
-	DeviceConnected <- true
 	helper.ChangeDeviceState("online", deviceID)
+	ss, err := p.DiscoverServices(nil)
+	if err != nil {
+		glog.Errorf("Failed to discover services, err: %s\n", err)
+	}
+	for _, s := range ss {
+		// Discovery characteristics
+		cs, err := p.DiscoverCharacteristics(nil, s)
+		if err != nil {
+			glog.Errorf("Failed to discover characteristics, err: %s\n", err)
```

Comment:
%s -> %v. also we can log the service name for which the characteristics could not be discovered.

Reply:
Done, thanks